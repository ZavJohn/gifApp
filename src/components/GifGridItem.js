import React  from 'react';
import PropTypes from 'prop-types';


export const GifGridItem = ( { id, title, url } ) => {
    return (
        <li key={ id } className="card animate__animated animate__flash">
            <img src={ url } alt={ title } />
            <p> { title }</p>
        </li>
    )
}

GifGridItem.propTypes = {
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};