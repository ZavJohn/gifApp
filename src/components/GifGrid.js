import React, { useEffect, useState } from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
    
    const { data:images, loading } = useFetchGifs( category );
    
        return (
            <>
                <h3 className="animate__animated animate__pulse"> { category } </h3>
                { loading && <p>Loading...</p> }
                <div className="card-grid">
                    <ul>
                        { 
                            images.map( img => (
                                <GifGridItem
                                    key={ img.id }
                                    { ...img }
                                />
                            ))
                        }
                    </ul>
                </div>
            </>
            
        )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};