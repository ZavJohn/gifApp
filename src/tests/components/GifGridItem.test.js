import React  from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
    test('Debe mostrar el componente correctamente', () => { 
        const id = '';
        const title = 'Esto es una prueba';
        const url = 'https://hechosparalavida.mx';
        const wrapper = shallow( <GifGridItem id={ id } title={ title } url={ url }/>);
        
        expect( wrapper ).toMatchSnapshot();
    });
});

