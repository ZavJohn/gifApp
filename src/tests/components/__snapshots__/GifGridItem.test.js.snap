// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<li
  className="card animate__animated animate__flash"
  key=""
>
  <img
    alt="Esto es una prueba"
    src="https://hechosparalavida.mx"
  />
  <p>
     
    Esto es una prueba
  </p>
</li>
`;
